### ISSUE TITLE
Assert when installing on or after non-version git tag

### ISSUE THREAD

**Issue by c1728p9** [pypa/setuptools-scm#62]
If I checkout an existing repo (In this case pyOCD) and create a non-version formatted tag, I get an assert when installing.  If I don't don't have a local tag then installing works correctly.  Is it a requirement when using setuptools_scm that the git repository can only have version formatted tags?

Machine: Windows 7 64 bit
setuptools-scm: Version 1.8.0
pyOCD commit: 87ca6b91477db7d458960504f2103dae9b7a319f

The sequence to reproduce is shown below.  If "git tag example_tag" is removed then the install succeeds.

```
cd c:\
git clone https://github.com/mbedmicro/pyOCD.git
cd pyOCD
git tag example_tag
virtualenv env
call env\Scripts\activate
pip install setuptools_scm
python setup.py install
```

Output:

```
Traceback (most recent call last):
  File "setup.py", line 66, in <module>
    include_package_data=True,  # include files from MANIFEST.in
  File "c:\python2.7.10_64\Lib\distutils\core.py", line 111, in setup
    _setup_distribution = dist = klass(attrs)
  File "c:\pyOCD\env\lib\site-packages\setuptools\dist.py", line 272, in __init__
    _Distribution.__init__(self,attrs)
  File "c:\python2.7.10_64\Lib\distutils\dist.py", line 287, in __init__
    self.finalize_options()
  File "c:\pyOCD\env\lib\site-packages\setuptools\dist.py", line 327, in finalize_options
    ep.load()(self, ep.name, value)
  File "c:\pyOCD\env\lib\site-packages\setuptools_scm\integration.py", line 19,in version_keyword
    dist.metadata.version = get_version(**value)
  File "c:\pyOCD\env\lib\site-packages\setuptools_scm\__init__.py", line 67, inget_version
    version = version_from_scm(root)
  File "c:\pyOCD\env\lib\site-packages\setuptools_scm\__init__.py", line 29, inversion_from_scm
    return ep.load()(root)
  File "c:\pyOCD\env\lib\site-packages\setuptools_scm\git.py", line 36, in parse
    return meta(tag, dirty=dirty, node=node)
  File "c:\pyOCD\env\lib\site-packages\setuptools_scm\version.py", line 85, in meta
    assert tag is not None, 'cant parse version %s' % tag
AssertionError: cant parse version None
```

**Comment by untitaker** [comment]
Yeah, that assumption is definetly made.

Not sure if we want to support anything else. Iterating through all tags if the current one isn't parsable seems like the obvious solution, OTOH this might unintentionally lead to some old version being determined if the current version tag is not properly formatted.

**Comment by c1728p9** [comment]
Good to know.  I'll stick to well formatted tags in pyOCD then :).

Iterating through the all the tags seems reasonable if it is something you decide to support.  Maybe you could check for the presence of a '.' character to tell if you can interpret the tag as a a version or a regular tag.  That way you would still get an error if there is a small typo when making a tag.

**Comment by RonnyPfannschmidt** [comment]
There is no reasonable support from the git side so for now we can't support it sanely

**Comment by c1728p9** [comment]
This should be a fairly easy change from the git side.  Just add a matching pattern for the version number to the command 'git describe'.  It would be a one line change here:
https://github.com/pypa/setuptools_scm/blob/master/setuptools_scm/git.py#L18

The full change would be something like:

```
out, err, ret = do_ex('git describe --dirty --tags --long --match "*.*"', root)
```

Or for more specific matching:

```
out, err, ret = do_ex('git describe --dirty --tags --long --match "v*.*.*"', root)
```

Let me know what you think.  If you would like, I can create a PR for this.

**Comment by RonnyPfannschmidt** [comment]
I see what you mean,that approach is indeed much more reasonable than the history scan,

The nonspecific match should be the default, and it should be possible to override it via a keyword argument

**Comment by RonnyPfannschmidt** [comment]
I would appreciate a PR, since I'm still waiting for my replacement computer

**Comment by RonnyPfannschmidt** [comment]
fixed in 25b0529 - will be released after a followup for passing in match expressions
